### PR TITLE
Added Supabase example using only javascript.

### PR DESCRIPTION
**Pull request by preetsuthar17**
I made simple example of using Supabase in normal JavaScript which basically takes input from user and saves it in database.

### CODE REVIEW

**Comment by Isaiah-Hamilton** [comment]
Kind of confused what the example exactly does.  can you create a `README.md` for what this example does/what its about and how to set it up.

**Comment by preetsuthar17** [comment]
> Kind of confused what the example exactly does. can you create a `README.md` for what this example does/what its about and how to set it up.

Yeahh done! i forgot to select that file earlier, my mistake.

File: examples/simple-username-database/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+
+
+# Supabase Project Documentation
+
+This documentation provides an overview of a JavaScript project that utilizes Supabase for the database. The project allows users to enter their username.It is an open-source project created by [preetsuthar17](https://github.com/preetsuthar17) and can be found on [GitHub](https://github.com/preetsuthar17).
```
Comment (gregnr) [inline examples/simple-username-database/README.md:5]: @preetsuthar17 ideally if we're bringing this example to Supabase we should keep the links relative to this example.

```suggestion
This documentation provides an overview of a JavaScript project that utilizes Supabase for the database. The project allows users to enter their username.
```

File: examples/simple-username-database/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+Before running the project, please make sure you have Node.js installed on your machine. If you don't have Node.js, you can download and install it from the official website: [Node.js Website](https://nodejs.org).
+
+## Installation
+
+To get started with the project, follow these steps:
+
+1. Clone the repository from GitHub:
+   ```
+   git clone https://github.com/preetsuthar17/supabase-gemo.git
```
Comment (gregnr) [inline examples/simple-username-database/README.md:20]: Here as well 👆

File: examples/simple-username-database/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+
+## Usage
+
+Once the project is set up, you can run it using the following command:
+
+```
+node index.js
+```
```
Comment (gregnr) [inline examples/simple-username-database/README.md:47]:
```suggestion
```shell
node index.js
```
```

File: examples/simple-username-database/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+[![Demo Vid]()](https://github.com/preetsuthar17/supabase-demo/assets/75468116/b5e88f63-525c-4211-8c50-b46e198d4632)
+
+
+![image](https://github.com/preetsuthar17/supabase-demo/assets/75468116/12f5a4e2-8a42-4b65-80d5-ede9161d6c0c)
```
Comment (gregnr) [inline examples/simple-username-database/README.md:52]: We should upload these to this repo if we plan to include them.

File: examples/simple-username-database/index.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+
+// one of the most simplest example of javascript using supabase datbase to take input from user and save it in database.
```
Comment (gregnr) [inline examples/simple-username-database/index.js:2]:
```suggestion
// Simple JavaScript example to take a username input and save it in a Supabase database.
```

File: examples/simple-username-database/index.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+
+// one of the most simplest example of javascript using supabase datbase to take input from user and save it in database.
+
+const { createClient } = require("@supabase/supabase-js");
+const readline = require("readline");
+
+const supabaseUrl = "replace-with-your-supabase-url";
+const supabaseKey =
+  "replace-with-your-supabasekey";
+const supabase = createClient(supabaseUrl, supabaseKey);
```
Comment (gregnr) [inline examples/simple-username-database/index.js:10]: We can probably combine these into one line for simplicity

File: examples/simple-username-database/index.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+
+// one of the most simplest example of javascript using supabase datbase to take input from user and save it in database.
+
+const { createClient } = require("@supabase/supabase-js");
+const readline = require("readline");
```
Comment (gregnr) [inline examples/simple-username-database/index.js:5]: Should we update to modern ES Module imports?

File: examples/simple-username-database/index.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    const { data, error } = await supabase
+      .from("usernames")
+      .insert({ name: name });
+
+    if (error) {
+      throw error;
+    }
+
+    console.log("Inserted name:", name);
+    import("chalk").then((chalk) => {
```
Comment (gregnr) [inline examples/simple-username-database/index.js:56]: Ideally we follow best practices by importing dependencies at the top.

File: examples/simple-username-database/package.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@supabase/supabase-js": "^2.26.0",
+    "chalk": "^5.2.0",
+    "readline": "^1.3.0"
+  }
+}
```
Comment (gregnr) [inline examples/simple-username-database/package.json:16]: Let's update these meta fields to match what we do in our other examples

**Review by gregnr — COMMENTED** [review]
Thanks for the PR @preetsuthar17. Left some comments/feedback for you.

**Comment by preetsuthar17** [comment]
It's all done! please check and if anything needs to be changed let me know. Some dependencies are just not working by including them on top.

File: examples/simple-username-database/README.md (hunk 2 above)
Comment (preetsuthar17) [reply examples/simple-username-database/README.md:20]: After merging pull request we can change that link to this repository.

File: examples/simple-username-database/README.md (hunk 4 above)
Comment (preetsuthar17) [reply examples/simple-username-database/README.md:51]: Alright, I'll add the image and video to the repository.

**Comment by gregnr** [comment]
Thanks for the updates @preetsuthar17. Taking a second look through this example - I think there's a few more things we should address:
- We should be clear (in the title / readme description) that this is a way to connect to Supabase from Node.js. I think indicating Node.js is key so that developers are clear on what it's intended for
- I'm hesitant to use 'usernames' as the table in this example, since usernames are usually tied closely with authentication which we don't include here. Could we come up with another entity that might be less confusing/less involved for developers?
- We should include the table schema in the README/migrations file so that developers can reproduce

**Comment by preetsuthar17** [comment]
can you suggest me name for the table?

**Comment by preetsuthar17** [comment]
I did minor changes in readme files and other files by changing the project name from username to name-reg. updated the readme and added the guide to generate the table in the dashboard using the SQL editor query.

**Comment by gregnr** [comment]
Thanks @preetsuthar17 - maybe instead of names we can store something like "todos"?

File: examples/simple-username-database/index.js (hunk 7 above)
Comment (gregnr) [reply examples/simple-username-database/index.js]: @preetsuthar17 thoughts on these other comments? I see we are still using `require()`.

**Comment by preetsuthar17** [comment]
That would be way better than names, I will change this for sure! i am busy for a while so I'll do it later.

**Comment by preetsuthar17** [comment]
@gregnr I changed the project name with it functionality tot todo

**Comment by preetsuthar17** [comment]
@gregnr All done! I added functionality to fetch data from the database and add data to the database via CMD. I also updated the table's name to `todoreg` and the rows in there!

**Comment by preetsuthar17** [comment]
@gregnr you here?

**Comment by gregnr** [comment]
Thanks @preetsuthar17. By chance did you see my comments earlier w.r.t. using `import` statements instead of `require()`? Also had another one about the meta fields in `package.json`. Let me know if you have any questions about those.

**Comment by preetsuthar17** [comment]
@gregnr I'm really sorry I have made updates to the imports and package.json file Please tell me if there is anything else that needs to be changed.

**Comment by preetsuthar17** [comment]
@gregnr is there anything else left to change?

**Pull request closed without merging**